docs/P4c-test-documentation.md

```diff
@@ -0,0 +1,143 @@
+# **Adding New Tests to the P4C Repository**
+
+**This document provides a guide for developers who want to add new tests to the P4C repository. It explains where to place test files, how to manage expected output files, and how to handle tests that are expected to fail (Xfail).**
+
+---
+
+## **1. Adding "Positive" Tests**
+
+Positive tests are those where the P4 compiler should process the input without any errors.
+
+### Steps for Adding Positive Tests
+
+1. **Source File Location**:
+
+   - Positive tests are located in multiple directories based on the purpose and the backend/tool being tested:
+
+     - `p4c/testdata/p4_16_samples`: General positive tests for the P4\_16 language.
+     - `p4c/testdata/p4_16_samples_outputs`: Expected outputs for these tests.
+     - Other directories may contain tests for specific backends or scenarios (e.g., BMv2 backend or PSA architecture).
```

Comment:
It would be really nice if we could be exhaustive and describe what every single directory under `p4c/testdata` is for, precisely.